plugin/builtin/buildbaron/build_baron.go

```diff
@@ -130,6 +150,44 @@ type searchReturnInfo struct {
 	Search string                  `json:"search"`
 }
 
+// raceSuggesters returns the JIRA ticket results from the altEndpoint suggester if it returns
+// within its configured interval, and returns the JIRA ticket results from the fallback suggester
+// otherwise.
+func raceSuggesters(fallback, altEndpoint suggester, t *task.Task) ([]thirdparty.JiraTicket, error) {
+	type result struct {
+		Tickets []thirdparty.JiraTicket
+		Error   error
+	}
+
+	// thirdparty/jira.go and thirdparty/http.go do not expose an API that accepts a context.Context.
+	fallbackCtx := context.TODO()
+	fallbackChan := make(chan result)
```

Comment:
Will you ever send more than 1 result into this channel? If not I'd make it a buffered channel with length 1. Also make sure to close the channel when you're done sending